jupyterhub/handlers/pages.py

```diff
@@ -118,6 +118,9 @@ def get(self, user_name=None, server_name=''):
 
     @needs_scope("servers")
     async def _get(self, user_name, server_name):
+        if '/' in server_name:
+            raise web.HTTPError(400, f"Invalid server_name: {server_name}")
```

Comment:
This check is only on GET not on POST, which is why the test is passing. The test doesn't use the single-server GET request, it users the _full user_ (all servers) GET request. It might make sense to move it to the centralized `BaseHandler.spawn_single_user`, or on the User's `new_spawner`. Since there are quite a few places that would also need this check, but they all go through `spawn_single_user` eventually.